File: pfm-sdk/src/main/java/com/tink/pfmsdk/mapper/AmountToDoubleConverter.java
package com.tink.pfmsdk.mapper;

import se.tink.core.models.misc.Amount;
import se.tink.modelConverter.AbstractConverter;

public class AmountToDoubleConverter extends AbstractConverter<Amount, Double> {

	@Override
	public Double convert(Amount source) {
		if (source.getValue() != null) {
			return source.getValue().doubleValue();
		}
		return 0d;
	}

	@Override
	public Class<Amount> getSourceClass() {
		return Amount.class;
	}

	@Override
	public Class<Double> getDestinationClass() {
		return Double.class;
	}

}

File: pfm-ui/src/main/java/com/tink/pfmui/mapper/AmountToDoubleConverter.kt
package com.tink.pfmui.mapper

import com.tink.model.misc.Amount
import se.tink.commons.extensions.doubleValue
import se.tink.modelConverter.AbstractConverter

internal class AmountToDoubleConverter :
    AbstractConverter<Amount, Double>() {
    override fun convert(source: Amount): Double {
        return source.value.doubleValue()
    }

    override fun getSourceClass(): Class<Amount> {
        return Amount::class.java
    }

    override fun getDestinationClass(): Class<Double> {
        return Double::class.java
    }
}